### PR TITLE
Fixes Media Daemon

### LINKED ISSUE #296: Media Discovery Daemon Not working
**Describe the bug**
USB skills not being created as the USB is inserted

**To Reproduce**
Run the install script and inserted USB

**Expected behavior**
Automatic USB skills are created

**Additional context**
I would like to work on it

### PR DESCRIPTION
**Pull request by stealthanthrax**
Fixes #296 

Error was caused by bad shebang statements and wrong paths

### CODE REVIEW

File: media_daemon/auto_skills.py [hunk 1]
```diff
@@ -27,8 +27,8 @@ def make_skill(): # pylint-enable
     for skill in skills:
         f.write(skill + '\n')
     f.close()
-    shutil.move( media_daemon_folder + 'custom_skill.txt', server_skill_folder)
-    f2 = open(server_settings_folder + 'customized_config.properties','a')
+    shutil.move( os.path.join(media_daemon_folder, 'custom_skill.txt'), server_skill_folder)
+    f2 = open(os.path.join(server_settings_folder, 'customized_config.properties'),'a')
```
Comment (hongquan) [inline media_daemon/auto_skills.py]: Better use the `with open()` syntax.

**Comment by stealthanthrax** [comment]
@hongquan , I figured out that the shell script is unable to execute the python script when it is executed through udev. Can I use pyudev instead of udev ??

**Comment by hongquan** [comment]
Which shell script and which python script?

**Comment by stealthanthrax** [comment]
The script `autostart.sh` is ran using udev , when the USB is inserted.Which in turn starts `auto_skills.py` . But if we execute the script independently , it works but not if started through UDEV

**Comment by stealthanthrax** [comment]
@hongquan , I have found the bug and will try to refactor it out

**Comment by stealthanthrax** [comment]
@hongquan , the GUI popup which starts is interfering with the script. As the rule has '99' digit in it making the rule of highest priority, the script is executed while the OS is mounting the USB and hence the code is unable to work. But if we make the number '99' any lower, the script doesn't even start because of low priority. 

Do you have any suggestions regarding it ??

**Comment by hongquan** [comment]
Which GUI? Isn't susi_linux targeted the headless Raspbian?

The number in the rule means: the lower number, the higher priority.

My suggestion is here: https://github.com/fossasia/susi_linux/issues/223#issue-338352255

> Use udisk to mount USB drive as non-root user.

and must run in headless Raspbian to prevent the disk from being [mounted automatically](https://github.com/fossasia/susi_linux/issues/223#issuecomment-402600426) by DE.

**Comment by stealthanthrax** [comment]
This PR is not relevant anymore

**Pull request closed without merging**